lib/report_formatter/timeline_message.rb

```diff
@@ -0,0 +1,84 @@
+include CompressedIds
+
+module ReportFormatter
+  class TimelineMessage
+    def initialize(rec, text, ems_cloud, ems_container)
+      @rec, @text, @ems_cloud, @ems_container = rec, text, ems_cloud, ems_container
+    end
+
+    def vm_name
+      "<a href='/vm/show/#{to_cid(@rec.vm_or_template_id)}'>#{@text}</a>" if @rec.vm_or_template_id
+    end
+
+    def src_vm_name
+      "<a href='/vm/show/#{to_cid(@rec.src_vm__id)}'>#{@text}</a>" if @rec.src_vm_id
```

Comment:
It's a verbatim copy of the original `elseif` code, I also do not know if it's good or bad - only that it is _exactly_ how it was before.